### PR TITLE
fix: handle tsconfig.json with comments

### LINKED ISSUE #2765: tsconfig.json with comments
The latest release on Deno panics when the tsconfig.json contains comments. While JSON is indeed invalid due to that reason, most people still uses it since it's how it comes by default on typescript.

```
deno --allow-net --config tsconfig.json server.ts
```
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ErrBox(DenoError { kind: InvalidInput, msg: "Compiler config is not a valid JSON" })', src\libcore\result.rs:999:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

```
cat tsconfig.json
```
```js
{
  "compilerOptions": {
    /* Basic Options */
    // "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
    "removeComments": true,                   /* Do not emit comments to output. */

    /* Strict Type-Checking Options */
    "strict": true,                           /* Enable all strict type-checking options. */
    // "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
    // "strictNullChecks": true,              /* Enable strict null checks. */
    // "strictFunctionTypes": true,           /* Enable strict checking of function types. */
    // "strictBindCallApply": true,           /* Enable strict 'bind', 'call', and 'apply' methods on functions. */
    // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
    // "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
    // "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */

    /* Additional Checks */
    "noUnusedLocals": true,                   /* Report errors on unused locals. */
    // "noUnusedParameters": true,            /* Report errors on unused parameters. */
    "noImplicitReturns": true,                /* Report error when not all code paths in function return a value. */
    "noFallthroughCasesInSwitch": true,       /* Report errors for fallthrough cases in switch statement. */
  }
}
```

Keep in mind, the trailing comma also causes the issue while working fine on older versions.

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes #2765

This PR removes parsing JSON file in Rust and instead uses regex to look for `"checkJs": true`.

Regex will capture following combinations:
```
"checkJs":true,

"checkJs": true,

"checkJs":
true,

"checkJs"
: true,
```

### CODE REVIEW

File: cli/compilers/ts.rs [hunk 1]
```diff
@@ -217,14 +213,8 @@ impl TsCompiler {
...
-    let config_json = config.json()?;
-    let compile_js = match &config_json.get("compilerOptions") {
-      Some(serde_json::Value::Object(m)) => match m.get("checkJs") {
-        Some(serde_json::Value::Bool(bool_)) => *bool_,
-        _ => false,
-      },
-      _ => false,
-    };
+    let config_str = config.string()?;
+    let compile_js = CHECK_JS_RE.is_match(&config_str);
```
Comment (ry) [inline cli/compilers/ts.rs]: Can you do a unit test here- showing comments in tsconfig rather than an integration test ?

Comment (bartlomieju) [reply cli/compilers/ts.rs]: You're right :+1: rewritten as unit tests

File: tests/038_checkjs_commented_json.tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (ry) [inline tests/038_checkjs_commented_json.tsconfig.json]: This file should be deleted, I think.

File: cli/compilers/ts.rs [hunk 3]
```diff
@@ -739,4 +726,42 @@ mod tests {
...
+      (
+        "",
+        false,
+      ),
+    ];
+
+    let path = temp_dir_path.join("tsconfig.json");
+    let path_str = path.to_str().unwrap().to_string();
+
+    for test_case in &test_cases {
```
Comment (ry) [inline cli/compilers/ts.rs]:
```rust
for (json_str, expected) in &test_cases {
  // ...
}
```

**Comment by ry** [comment]
Sorry if this is a dumb question, but why do we need to have check_js extracted again?

**Comment by bartlomieju** [comment]
@ry https://github.com/denoland/deno/issues/2765#issuecomment-520794508

**Pull request merged**